main.tf

```diff
@@ -74,6 +74,34 @@ locals {
       }
     ]
   }
+
+  # Origin Shield mapping configuration
+  # See: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/origin-shield.html
+  origin_shield_region_mapping = {
+    # Regions where Origin Shield is available
+    us-east-2      = "us-east-2"      # US East (Ohio)
+    us-east-1      = "us-east-1"      # US East (N. Virginia)
+    us-west-2      = "us-west-2"      # US West (Oregon)
+    ap-south-1     = "ap-south-1"     # Asia Pacific (Mumbai)
+    ap-northeast-2 = "ap-northeast-2" # Asia Pacific (Seoul)
+    ap-southeast-1 = "ap-southeast-1" # Asia Pacific (Singapore)
+    ap-southeast-2 = "ap-southeast-2" # Asia Pacific (Sydney)
+    ap-northeast-1 = "ap-northeast-1" # Asia Pacific (Tokyo)
+    eu-central-1   = "eu-central-1"   # Europe (Frankfurt)
+    eu-west-1      = "eu-west-1"      # Europe (Ireland)
+    eu-west-2      = "eu-west-2"      # Europe (London)
+    sa-east-1      = "sa-east-1"      # South America (São Paulo)
+
+    # Regions where Origin Shield is NOT available (choose closest region)
+    us-west-1    = "us-west-2"      # US West (N. California)
+    af-south-1   = "eu-west-1"      # Africa (Cape Town)
+    ap-east-1    = "ap-southeast-1" # Asia Pacific (Hong Kong)
+    ca-central-1 = "us-east-1"      # Canada (Central)
+    eu-south-1   = "eu-central-1"   # Europe (Milan)
+    eu-west-3    = "eu-west-2"      # Europe (Paris)
+    eu-north-1   = "eu-west-2"      # Europe (Stockholm)
+    me-south-1   = "ap-south-1"     # Middle East (Bahrain)
+  }
```

Comment:
See my other comment on this type of behaviour not having much precedent in our modules.

Reply:
No, we do similar things here: https://github.com/cloudposse/terraform-aws-utils/blob/master/main.tf

Reply:
ironically, I see we are using a `local` for it too, but I think a `var` would be more appropriate in this case.

Reply:
@osterman right, but I didn't think https://github.com/cloudposse/terraform-aws-utils/blob/master/main.tf of a proper example since it's a utility module specifically meant to map region short-hands and not a regular module. But anyways, if it's acceptable precedent in your view, then that helps.